file-re-encryption-processor/src/test/java/uk/ac/ebi/ega/file/re/encryption/processor/services/ReEncryptServiceTest.java

```diff
@@ -0,0 +1,126 @@
+    private static final String BASE_PATH = "src/test/resources/keyPairTest/";
+    private static final String JOB_NAME = "re-encrypt-job";
+    private static final String REENCRYPT_FILE_NAME = "test_file_reencrypt.txt.gpg";
+    private static final String PASSWORD = "test";
+
+    @Before
+    public void init() {
+
+        reEncryptJobParameterService = new ReEncryptJobParameterService(reEncryptParametersRepository);
+        MockitoAnnotations.initMocks(reEncryptJobParameterService);
+
+        final IMailingService mailingService = Mockito.mock(IMailingService.class);
+        final Job<ReEncryptJobParameters> job = new ReEncryptJob(dosId -> {
+            return new LocalStorageFile("b2e6283b2044de260d6df0e854cd3fa2", BASE_PATH + "test_file.txt.gpg");
+        }, PASSWORD.toCharArray());
+
+        final KafkaTemplate<String, ReEncryptComplete> kafkaTemplate = Mockito.mock(KafkaTemplate.class);
+        final ReEncryptPersistenceService reEncryptPersistenceService = new ReEncryptPersistenceService(jobExecutionRepository, jobRunRepository, EMPTY_STRING, reEncryptJobParameterService);
+        reEncryptService = new ReEncryptService(reEncryptPersistenceService, mailingService, EMPTY_STRING, job, kafkaTemplate, EMPTY_STRING);
+        MockitoAnnotations.initMocks(reEncryptService);
+    }
+
+    @Test
+    public void createJob_SuppliedCorrectArguments_ExecutesSuccessfully() {
+
+        when(jobExecutionRepository.save(any(JobExecutionEntity.class))).thenReturn(new JobExecutionEntity());
+
+        final Optional<JobExecution<ReEncryptJobParameters>> jobExecutionOptional = reEncryptService.createJob(EMPTY_STRING, EMPTY_STRING, EMPTY_STRING, new char[0]);
+
+        assertTrue(jobExecutionOptional.isPresent());
+    }
+
+    @Test
+    public void reEncrypt_SuppliedCorrectArgument_ExecutesSuccessfully() {
+
+        final ReEncryptJobParameters reEncryptJobParameters = new ReEncryptJobParameters(EMPTY_STRING, new File(BASE_PATH + REENCRYPT_FILE_NAME).getAbsolutePath(), PASSWORD.toCharArray());
+        final JobExecution<ReEncryptJobParameters> jobExecution = new JobExecution(EMPTY_STRING, JOB_NAME, reEncryptJobParameters);
+
+        when(jobRunRepository.save(any(JobRun.class))).thenReturn(new JobRun());
+
+        final Result result = reEncryptService.reEncrypt(jobExecution);
+
+        assertEquals(Result.Status.SUCCESS, result.getStatus());
+    }
+
+    @Test
+    public void getUnfinishedJob_ExecutesSuccessfullyAndReturnsUnfinishedJob() {
+
+        when(jobExecutionRepository.findByInstanceId(anyString())).thenReturn(Optional.ofNullable(new JobExecutionEntity(EMPTY_STRING, EMPTY_STRING, EMPTY_STRING)));
+
+        final ReEncryptParametersEntity reEncryptParametersEntity = new ReEncryptParametersEntity(EMPTY_STRING, EMPTY_STRING, EMPTY_STRING, EMPTY_STRING);
+
+        when(reEncryptParametersRepository.findById(anyString())).thenReturn(Optional.ofNullable(reEncryptParametersEntity));
```

Comment:
```suggestion
        when(reEncryptParametersRepository.findById(anyString())).thenReturn(Optional.of(reEncryptParametersEntity));
```

Because we know that the `reEncryptParametersEntity` is not null (plus then IntelliJ doesn't complain).

Reply:
Yes. I did the changes accordingly.